cmd/zc_filter.go

```diff
@@ -146,6 +146,24 @@ func (f *includeFilter) doesPass(storedObject storedObject) bool {
 	return false
 }
 
+// getEnumerationPreFilter returns a prefix, if any, which can be used service-side to pre-select
+// things that will pass the filter. E.g. if there's exactly one include pattern, and it is
+// "foo*bar", then this routine will return "foo", since only things starting with "foo" can pass the filters.
+// Service side enumeration code can be given that prefix, to optimize the enumeration.
+func (f *includeFilter) getEnumerationPreFilter() string {
```

Comment:
Actually, I redact my approval.

Our include filter isn't repeatedly added. It's a single filter, with multiple potential patterns.

For instance, if I supplied

a*;b*

I suspect this might introduce a bug where none of the b* entries are found.

Reply:
Yes. I ignored the potential optimization on purpose in that case.  I felt I couldn't safely code and test that more complex version in the time available.  (I think I even left a comment in the code saying I'd deliberately not done it). 
We could add it to this code when we cherry pick this into 10.4, but I'm not comfortable with adding the complexity of that to this PR - given that the customer we are doing this PR for is time-sensitive, and has no need for a multi-part filter.
@adreed-msft Can you reconsider your approval in the light of the above?

Reply:
In this kind of scenario, I'd definitely agree to dumb-down the logic as much as possible to avoid potential bugs.

Reply:
Yeah, I'll reconsider approval after I look at the new push. I suspect with our discussion lastnight, I'll be happy with this